lib/util/redgif.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:http/http.dart' as http;
+import 'package:path/path.dart';
+
+bool isRedGif(String? url) {
+  return url != null && url.contains('redgifs.com');
+}
+
+//TODO Get a real API Key
+//TODO Cache this value (looks to be good for about a day)
+Future<String> _getAuthtoken() async {
+  final response =
```

Comment:
you can just make this a static/global variable so that until proper caching is added it will _at least_ last the lifetime of the app.